### PR TITLE
Add support for PyPy 3.10

### LINKED ISSUE #112: pypy3.10: AttributeError: 'SSLContext' object has no attribute '_ctx'. Did you mean: 'ctx'?
When running the test suite on pypy3.10 7.3.12, every test fails with the following error:

```pytb
_____________________________________________________ test_verify_mode_cert_none ______________________________________________________

    def test_verify_mode_cert_none():
>       ctx = truststore.SSLContext(ssl.PROTOCOL_TLS_CLIENT)

tests/test_sslcontext.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib/pypy3.10/ssl.py:496: in __new__
    self = _SSLContext.__new__(cls, protocol)
/usr/lib/pypy3.10/_cffi_ssl/_stdssl/__init__.py:1221: in __new__
    lib.SSL_CTX_set_post_handshake_auth(self.ctx, self.post_handshake_auth)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <truststore._api.SSLContext object at 0x00005599f6700138>

    @property
    def post_handshake_auth(self) -> bool:
>       return self._ctx.post_handshake_auth
E       AttributeError: 'SSLContext' object has no attribute '_ctx'. Did you mean: 'ctx'?

.nox/test/lib/pypy3.10/site-packages/truststore/_api.py:232: AttributeError
```

### PR DESCRIPTION
**Pull request by sethmlarson**
This is very much a hack, I don't like it, but attempting to add support to PyPy while maintaining the `SSLContext` subclassing is a huge pain from my initial attempt. The `__class__` hack maintains compatibility for aiohttp and trustme.

Closes #112

### CODE REVIEW

**Comment by davisagli** [comment]
Oh my. Mostly I'm just impressed if it works. Good call on only changing it for non-CPython implementations, in case someone is already using isinstance to check if they've got our SSLContext.

I'll wait for final review until the tests pass.

**Comment by sethmlarson** [comment]
Looks like a bug in PyPy's `SSLContext.load_cert_chain()` function, they're not accepting `PosixPath` objects. Unfortunately their GitLab instance appears to be offline right now. I guess I'll check later?

**Comment by sethmlarson** [comment]
Okay @davisagli tests are passing so ready for review. If you'd like to take a stab at PyPy support without the horrible hacks please do, I would like to not have them :sweat:

**Review by davisagli — DISMISSED** [review]

**Review by davisagli — APPROVED** [review]

File: tests/conftest.py
Comment (mattip) [inline tests/conftest.py:149]: Sorry about that, is there a simple reproducer for this? PyPy passes the CPython ssl stdlib tests, maybe there is a case missing there?

**Comment by mattip** [comment]
Thanks for getting PyPy to work. I am not familiar with the project, but am a PyPy core dev. Please ping me (mattip) if you come across any other PyPy problems.

**Comment by mattip** [comment]
I opened https://foss.heptapod.net/pypy/pypy/-/issues/4002 to continue the discussion.

**Comment by sethmlarson** [comment]
@mattip Thanks Matti! We'll keep you in mind if we run into future problems :)

**Pull request merged**
